src/cellularnoise.rs

```diff
@@ -28,53 +28,53 @@ fn noise_gen(prec_as_str : &str, smoothing_level_as_str : &str, birth_limit_as_s
     for _timer in 0..smoothing_level {
         let zplane_old = zplane.clone();
         for i in 0..zplane_old.len() {
-            for j in 0..zplane_old.len(){
+            for j in 0..zplane_old[i].len(){
                 let mut sum = 0;
 
                 if j > 0{
-                    sum += if zplane_old[j-1][i] {1} else {0};
+                    sum += if zplane_old[i-1][j] {1} else {0};
                 }
 
                 if i > 0 {
-                    sum += if zplane_old[j][i-1]  {1} else {0};
+                    sum += if zplane_old[i][j-1]  {1} else {0};
                 }
 
                 if i > 0 && j > 0 {
-                    sum += if zplane_old[j-1][i-1] {1} else {0};
+                    sum += if zplane_old[i-1][j-1] {1} else {0};
                 }
 
-                if j < zplane_old.len()-1 {
-                    sum += if zplane_old[j+1][i] {1} else {0};
+                if j < zplane_old[i].len()-1 {
+                    sum += if zplane_old[i+1][j] {1} else {0};
                 }
 
                 if i < zplane_old.len()-1 {
-                    sum += if zplane_old[j][i+1] {1} else {0};
+                    sum += if zplane_old[i][j+1] {1} else {0};
                 }
 
                 if i < zplane_old.len()-1 && j < zplane_old.len()-1{
-                    sum += if zplane_old[j+1][i+1] {1} else {0};
+                    sum += if zplane_old[i+1][j+1] {1} else {0};
                 }
 
-                if i > 0 && j < zplane_old.len()-1 {
-                    sum += if zplane_old[j+1][i-1] {1} else {0};
+                if i > 0 && j < zplane_old[i].len()-1 {
+                    sum += if zplane_old[i+1][j-1] {1} else {0};
                 }
 
                 if j > 0 && i < zplane_old.len()-1 {
-                    sum += if zplane_old[j-1][i+1] {1} else {0};
+                    sum += if zplane_old[i-1][j+1] {1} else {0};
                 }
 
-                if zplane_old[i][i] == true{
+                if zplane_old[i][j] == true{
                     if sum < death_limit{
-                        zplane[j][i] = false;
+                        zplane[i][j] = false;
                     } else{
-                        zplane[j][i] = true;
+                        zplane[i][j] = true;
                     }
                 }
                 else{
                     if sum > birth_limit{
-                        zplane[j][i] = false;
+                        zplane[i][j] = false;
```

Comment:
I think you have false/true backwards on this one

Reply:
yep he does

Reply:
that might be the case